src/fs/at.rs

```diff
@@ -319,7 +315,31 @@ pub fn utimensat<P: path::Arg, Fd: AsFd>(
 #[inline]
 #[doc(alias = "fchmodat")]
 pub fn chmodat<P: path::Arg, Fd: AsFd>(dirfd: Fd, path: P, mode: Mode) -> io::Result<()> {
-    path.into_with_c_str(|path| backend::fs::syscalls::chmodat(dirfd.as_fd(), path, mode))
+    chmodat_with(dirfd, path, mode, AtFlags::empty())
+}
+
+/// `fchmodat(dirfd, path, mode, flags)`—Sets file or directory permissions.
+///
+/// Platform support for flags varies widely, for example Linux does not yet
+/// implement `AT_SYMLINK_NOFOLLOW` (except when using the libc backend with musl libc)
+/// and will return `EINVAL` with non-empty flags.
```

Comment:
Glibc and musl use `EINVAL` for any flag other than `AT_SYMLINK_NOFOLLOW`, but they use `EOPNOTSUP` for `AT_SYMLINK_FOLLOW` if they can't emulate it, so it looks like `EOPNOTSUP` is the way to go here.

Reply:
Ah, interesting, [true](https://github.com/lattera/glibc/blob/895ef79e04a953cac1493863bcae29ad85657ee1/sysdeps/unix/sysv/linux/fchmodat.c#L36). Changed.